pulp_puppet_plugins/pulp_puppet/plugins/importers/importer.py

```diff
@@ -80,9 +81,21 @@ def upload_unit(self, repo, type_id, unit_key, metadata, file_path, conduit, con
         try:
             report = upload.handle_uploaded_unit(repo, type_id, unit_key, metadata, file_path,
                                                  conduit)
+
```

Comment:
In both of these cases, what would happen if you just raised a ``PulpCodedException`` with an appropriate message and let it bubble up? I think that's the ideal pattern at this point, but perhaps there are some challenges doing that here?